exporter/containerimage/export.go

```diff
@@ -108,6 +113,7 @@ func (e *imageExporterInstance) Export(ctx context.Context, src exporter.Source)
 	for k, v := range e.meta {
 		src.Metadata[k] = v
 	}
+	ctx = namespaces.WithNamespace(ctx, e.namespace)
```

Comment:
I think your CI failure is because if the caller didn't pass `keyNamespace` then this (and the other) will nuke the default namespace (whatever that is) with `""`.

Reply:
Didn't test this but, looking at the snapshotter/contentstore used they seem to ignore the value from context, so I guess atm this new namespace only applies to the `imagestore.Update` call. I would think that the blobs also need to exist in the same namespace before the `Update` call. If so this will need some changes in the `Commit()` logic to handle different namespace for source blobs/snapshots and target blobs(or containerd change to reflink blobs automatically in this case). If not then yes the empty string could be the only issue. When this is fixed we should have an integration test as well for this new field.

Reply:
so the thing is I was setting a namespace in the buildkit config and I got a containerd error thrown on the call to export the image back to containerd, that the namespace was undefined

Reply:
so could this be a different bug? because i would prefer if I didn't have to set it this way and it would use the one i had defined in the buildkit containerd config

Reply:
@jessfraz So is this fixing an issue when using a custom global namespace for containerd, or are you trying to  set a different namespace for the exported image than what buildkit is regularly using? I initally thought the latter but if not then it indeed should be some other bug. The `ImageStore` should already have the default namespace defined in the worker and there shouldn't be a need to override in exporter.

Reply:
I just want to use the one I set in the buildkit config for containerd :)

Reply:
ah ok I will try to figure out the bug then :)

Reply:
btw, couldn't reproduce:

```
> buildkitd --oci-worker=false --containerd-worker-namespace=foobar
> buildctl build --frontend=dockerfile.v0 --local context=. --local dockerfile=. --exporter=image --exporter-opt name=docker.io/tonistiigi/test
...
> ctr --namespace=foobar image ls

REF                       TYPE                                                 DIGEST                                                                  SIZE    PLATFORMS   LABELS
docker.io/tonistiigi/test application/vnd.docker.distribution.manifest.v2+json sha256:08a5efa865b8ab44f611ed9c900eb45b8aae3f7ed222c9aae2549cf1b129b39e 2.1 MiB linux/amd64 -

```

Reply:
omg ok its me... will try again so sorry